src/transforms/tag_cardinality_limit.rs

```diff
@@ -0,0 +1,459 @@
+use super::Transform;
+use crate::{
+    topology::config::{DataType, TransformConfig, TransformContext, TransformDescription},
+    Event,
+};
+use bloom::{BloomFilter, ASMS};
+use serde::{Deserialize, Serialize};
+use std::collections::{HashMap, HashSet};
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+// TODO: add back when serde-rs/serde#1358 is addressed
+//#[serde(deny_unknown_fields)]
+pub struct TagCardinalityLimitConfig {
+    #[serde(default = "default_value_limit")]
+    pub value_limit: u32,
+
+    #[serde(default = "default_limit_exceeded_action")]
+    pub limit_exceeded_action: LimitExceededAction,
+
+    #[serde(flatten)]
+    pub mode: Mode,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(tag = "mode", rename_all = "snake_case", deny_unknown_fields)]
+pub enum Mode {
```

Comment:
So I don't actually think mode="hash_set" and mode="bloom_filter" are great names for the config options as they seem to be too focused on implementation detail instead of behavior. 
 The best alternative I could come up with was "reliable" vs "efficient", though I don't love the resulting implication that the bloom filter approach is "unreliable"...  Open to input if anyone has ideas for better names here!

Reply:
Yes those are much better names indeed, thanks!